R/formatter.r

```diff
@@ -760,56 +757,40 @@ pvalue <- function(x, accuracy = .001, decimal.mark = ".", add_p = FALSE) {
 #' number_bytes_format()(sample(3000000000, 10))
 #' number_bytes(sample(3000000000, 10))
 #' number_bytes(sample(3000000000, 10), accuracy = .1)
+#' number_bytes(1024^(0:4))
+#' number_bytes(1024^(0:4), units = "si", accuracy = .01)
+#' number_bytes(1000^(1:3), "kB", units = "si")
 number_bytes_format <- function(symbol = "auto", units = "binary", ...) {
   function(x) number_bytes(x, symbol, units, ...)
 }
 
 #' @export
 #' @rdname number_bytes_format
 number_bytes <- function(x, symbol = "auto", units = c("binary", "si"), ...) {
-  symbols <- c(
-    "auto",
-    "b", "Kb", "Mb", "Gb", "Tb", "Pb", "Eb", "Zb", "Yb",
-    "B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB",
-    "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"
-  )
+  units <- match.arg(units, c("binary", "si"))
+
+  powers <- si_powers[si_powers >= 3] / 3 # powers of 1000
+  prefix <- names(powers)
+
+  symbols <- c("auto", "B", switch(units,
+    si     = paste0(prefix, "B"),
+    binary = paste0(toupper(prefix), "iB")
+  ))
 
   if (!(symbol %in% symbols)) {
-    warning(paste(
-      "`symbol` must be one of '", paste0(symbols, collapse = "', '"),
-      "'. Defaulting to 'auto'."
-    ), call. = F)
+    warning("`symbol` must be one of: '", paste0(symbols, collapse = "', '"),
```

Comment:
Can you please follow the long-lines style: http://style.tidyverse.org/syntax.html#long-lines